final/problem-def-ana/problem-ana.tex

```diff
@@ -92,30 +74,39 @@ \subsubsection{Language specific additional commands}
 language is a DSL. It is clear therefore that a different approach
 should be considered.
 
-One possible solution is to allow for language specific configurations
-that are loaded with the language definition of that language. This
-can be done by extending the editor services discussed in
-\cref{ssec:editor-serv} with REPL commands: similar to menu actions,
-where one can define menu button ``Run'' to run a program, one may
-define a ``load-file'' command and bind it to an action.
+One possible solution is to allow for configurations of
+language-specific commands, that are loaded with the language
+definition of that language. This can be done by extending the editor
+services discussed in \cref{ssec:editor-serv} with REPL
+commands. Similar to menu actions, where one can define menu buttons
+such as ``Run'' to run a program, one may then even bind the
+``load-file'' command to an action.
 
-\subsubsection{Redefining terms bound to names}
-\label{sec:redef-cont-bound}
-When the user is prototyping methods inside the REPL, they would
-likely want to be able to redefine that method to be of a different
-implementation. However, this poses a similar problem as in the
-previous section: for some languages it may not be possible nor
-desirable to do so outside of the context of a REPL. Thus requiring
-the language designer to extend the language with such abilities is
-again an inadequate solution.
+\subsection{REPL-specific semantics}
+\label{sec:repl-spec-semant}
+Sometimes a property of some language stands in the way of the rapid
+prototyping capabilities that are desirable for a REPL. For example, a
+language like Haskell does not allow for redefining a function's
+implementation in its normal semantics, whereas in the context of a
+REPL this is usually what one wants.
```

Comment:
one usually wants the the language to allow redefining in the context of a REPL.